### PR TITLE
Create systemd setup for backend

### PR DESCRIPTION
**Pull request by decentralizedb**
If anyone wants to create a system service to run teos as a system service to start on the background, the instructions to create it are described in a new section in the install.md file.

### CODE REVIEW

**Review by sr-gi — COMMENTED** [review]
Hey @decentralizedb, thanks for taking care of this.

I think this should be part of INSTALL.md instead of its own file.

It'll also be nice for this PR to have a minimal description.

**Comment by meryacine** [comment]
Is `bitcoind.service` installed by default with bitcoin core?
If not I think you should publish it too (or put a hyperlink to it).

File: INSTALL.md [hunk 1]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+# Systemd setup for backend
+
+
+#### Create a simple file copying the contents below to the file, and use the command:
+`sudo nano /etc/systemd/system/teosd.service`
+#### This will create a file named teos.service in the /etc/systemd/system/ path
+
+#### One you create it and copy and paste, change the information with your proper installation path, user, and comments:
+
+#### ------------- don't copy this line, copy under it --------------------------
```
Comment (sr-gi) [inline INSTALL.md:32]: I don't think this wrapping is really necessary. Given the file content is already formatted using the GH code formatter (```) it is pretty clear what needs to go in the file.

File: INSTALL.md [hunk 2]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+PrivateDevices=true
+
+# Deny the creation of writable and executable memory mappings.
+MemoryDenyWriteExecute=true
+
+[Install]
+WantedBy=multi-user.target
+
+```
+#### ------------- don't copy this line, copy above it --------------------------
```
Comment (sr-gi) [inline INSTALL.md:79]: Remove

File: INSTALL.md [hunk 3]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
 cargo test
 ```
 
-Please refer to the cargo documentation for more detailed instructions.
\ No newline at end of file
+Please refer to the cargo documentation for more detailed instructions.
+
+
+### If you want to create a system service to run your application follow the instructions below:
```
Comment (sr-gi) [inline INSTALL.md:20]: I'd move this inside the `Systemd setup for backend` section.

Also, If you want -> Should you want to / If you would like to

File: INSTALL.md [hunk 4]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+
+
+# Systemd setup for backend
+
+
+#### Create a simple file copying the contents below to the file, and use the command:
```
Comment (sr-gi) [inline INSTALL.md:26]: Using you're preferred text editor, create a file in the `systemd` folder. For instance:

`sudo nano /etc/systemd/system/teosd.service`

File: INSTALL.md [hunk 5]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+`sudo nano /etc/systemd/system/teosd.service`
+#### This will create a file named teos.service in the /etc/systemd/system/ path
+
+#### One you create it and copy and paste, change the information with your proper installation path, user, and comments:
+
```
Comment (sr-gi) [inline INSTALL.md:31]: Copy the following information replacing the relevant information with yours (`<USER>` and `<GROUP>`):

File: INSTALL.md [hunk 6]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+# Deny the creation of writable and executable memory mappings.
+MemoryDenyWriteExecute=true
+
+[Install]
+WantedBy=multi-user.target
+
+```
+#### ------------- don't copy this line, copy above it --------------------------
+
+#### The next step is to enable the service with the following command:
```
Comment (sr-gi) [inline INSTALL.md:81]: [...] is enabling the service. You can do so by running:

File: INSTALL.md [hunk 7]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+`sudo systemctl enable teosd.service`
+
+#### And start the service with the command:
```
Comment (sr-gi) [inline INSTALL.md:84]: Finally, you can start the service by running:

File: INSTALL.md [hunk 8]
```diff
@@ -14,4 +14,75 @@ You can run tests with:
...
+`sudo systemctl start teosd.service`
+
+#### If you need to stop the service, use the command:
+`sudo systemctl stop teosd.service`
```
Comment (sr-gi) [inline INSTALL.md:88]: I don't think this is really necessary

**Review by sr-gi — CHANGES_REQUESTED** [review]
Some nits.

As a general comment, I don't think the text needs to be bolded for all paragraphs (####).

**Comment by sr-gi** [comment]
> 

Agreed, if this depends on another service it would be nice to have a link to it from a well know guide (such as raspibolt)

**Comment by sr-gi** [comment]
> Is `bitcoind.service` installed by default with bitcoin core? If not I think you should publish it too (or put a hyperlink to it).

Agreed, if this depends on another service it would be nice to have a link to it from a well know guide (such as raspibolt)

File: INSTALL.md (hunk 1 above)
Comment (decentralizedb) [reply INSTALL.md]: Ok, perfect, I removed wrapping and incorporated the parameters for logging in the system services file, please review it.

File: INSTALL.md (hunk 2 above)
Comment (decentralizedb) [reply INSTALL.md]: Done

File: INSTALL.md (hunk 4 above)
Comment (decentralizedb) [reply INSTALL.md]: All formatting and wording comments have been modified.

File: INSTALL.md (hunk 6 above)
Comment (decentralizedb) [reply INSTALL.md]: Done

File: INSTALL.md (hunk 3 above)
Comment (decentralizedb) [reply INSTALL.md]: Done

File: INSTALL.md (hunk 5 above)
Comment (decentralizedb) [reply INSTALL.md]: Done

File: INSTALL.md (hunk 7 above)
Comment (decentralizedb) [reply INSTALL.md]: Done

File: INSTALL.md (hunk 8 above)
Comment (decentralizedb) [reply INSTALL.md]: Removed

**Comment by decentralizedb** [comment]
> Is `bitcoind.service` installed by default with bitcoin core? If not I think you should publish it too (or put a hyperlink to it).

I added the link to a guide that is quite detailed and updated instructions. 
This guide has just been released and is quite good in and installation oriented on how to set up on most hardware platforms because it only uses standard Debian Linux commands, as the guide indicates.

**Comment by decentralizedb** [comment]
I attended all indications, I hope this time it will be ok, but if not, please let me know. I'm happy to help and learn!  xD

**Comment by sr-gi** [comment]
Tagging @openoms here since he's way more knowledgeable than me

**Review by sr-gi — CHANGES_REQUESTED** [review]
I think this makes sense, but I'd structure it differently. I think we can follow the same structure followed by Bitcoin Core for their init files (check https://github.com/bitcoin/bitcoin/blob/master/contrib/init and https://github.com/bitcoin/bitcoin/blob/master/contrib/init/bitcoind.service), that is:

- [ ]  Create a new folder in the project root called `contrib`
    - [ ] Create a subfolder called `init`
    - [ ] Create a `README.md` with most of the content added to this file in the PR
    - [ ] Create a `teosd.service` file including the relevant service information
- [ ] Link the `contrib/init/README.md` file to `INSTALL.md` under the `Systemd setup for backend`

This may like:

```
[...]
Please refer to the cargo documentation for more detailed instructions.

# Systemd setup for backend
Refer to init(LINK) for a detailed explanation of how to set up your systemd service or something on these lines.
```

**Pull request closed without merging**